src/AutoRest.CSharp.V3/Utilities/StringExtensions.cs

```diff
@@ -17,14 +15,54 @@ internal static class StringExtensions
         public static bool IsNullOrEmpty(this string? text) => String.IsNullOrEmpty(text);
         public static bool IsNullOrWhiteSpace(this string? text) => String.IsNullOrWhiteSpace(text);
 
-        [return: NotNullIfNotNull("text")]
-        private static string? RemoveNonWordCharacters(this string? text) => !text.IsNullOrEmpty() ? Regex.Replace(text, @"\W+", String.Empty) : text;
-        [return: NotNullIfNotNull("text")]
-        private static string? PrependUnderscoreIfNumbers(this string? text) => Regex.IsMatch(text ?? String.Empty, @"^\d") ? $"_{text}" : text;
         [return: NotNullIfNotNull("name")]
-        public static string ToCleanName(this string name) => name.ToPascalCase().RemoveNonWordCharacters().PrependUnderscoreIfNumbers();
+        public static string ToCleanName(this string name, bool camelCase = true)
```

Comment:
So, in what cases are we cleaning names? Because, we should be getting 'good' names from AutoRest, right? So, I'm trying to understand why we still have custom name cleaning. We can use the `naming:` node in our config:
https://github.com/Azure/autorest.modelerfour/tree/master/modelerfour#modelfour-options

Reply:
Response headers is the main one.

Reply:
Filed https://github.com/Azure/autorest.modelerfour/issues/262